src/cmd/flux-jobs.py

```diff
@@ -0,0 +1,202 @@
+    parser.add_argument(
+        "-a", action="store_true", help="List all jobs for current user"
+    )
+    # -A equivalent to -s "pending,running,inactive" and -u set to "all"
+    parser.add_argument("-A", action="store_true", help="List all jobs for all users")
+    parser.add_argument(
+        "-c",
+        "--count",
+        type=int,
+        metavar="N",
+        default=1000,
+        help="Limit output to N jobs(default 1000)",
+    )
+    parser.add_argument(
+        "-s",
+        "--states",
+        type=str,
+        metavar="STATES",
+        default="pending,running",
+        help="List jobs in specific states(pending,running,inactive)",
+    )
+    parser.add_argument(
+        "--suppress-header",
+        action="store_true",
+        help="Suppress printing of header line",
+    )
+    parser.add_argument(
+        "-u",
+        "--userid",
+        type=int,
+        metavar="UID",
+        default=os.geteuid(),
+        help='Limit output to specific userid (Specify "all" for all users)',
+    )
+    parser.add_argument(
+        "-o",
+        "--format",
+        type=str,
+        metavar="FORMAT",
+        help="Specify output format using Python f-string formation",
+    )
+
+    args = parser.parse_args()
+
+    h = flux.Flux()
+
+    # Future optimization, reduce attrs based on what is in output
+    # format, to reduce potentially large return RPC.
+    attrs = [
+        "userid",
+        "priority",
+        "state",
+        "job-name",
+        "task-count",
+        "t_submit",
+        "t_depend",
+        "t_sched",
+        "t_run",
+        "t_cleanup",
+        "t_inactive",
+    ]
+
+    if args.a:
+        args.userid = os.geteuid()
+    if args.A:
+        args.userid = flux.constants.FLUX_USERID_UNKNOWN
+    if args.a or args.A:
+        args.states = "pending,running,inactive"
+
+    flags = 0
+    for state in args.states.split(","):
+        if state.lower() == "pending":
+            flags |= flux.constants.FLUX_JOB_LIST_PENDING
+        elif state.lower() == "running":
+            flags |= flux.constants.FLUX_JOB_LIST_RUNNING
+        elif state.lower() == "inactive":
+            flags |= flux.constants.FLUX_JOB_LIST_INACTIVE
+        else:
+            print("Invalid state specified", file=sys.stderr)
+            sys.exit(1)
+
+    if flags == 0:
+        flags |= flux.constants.FLUX_JOB_LIST_PENDING
+        flags |= flux.constants.FLUX_JOB_LIST_RUNNING
+
+    rpc_handle = flux.job.job_list(h, args.count, attrs, args.userid, flags)
+    try:
+        jobs = flux.job.job_list_get(rpc_handle)
+    except EnvironmentError as e:
+        print("{}: {}".format("rpc", e.strerror), file=sys.stderr)
+        sys.exit(1)
+
+    if args.format:
+        output_format(args.format, jobs)
+    else:
+        if not args.suppress_header:
+            print(
+                "%s\t\t%-8.8s %-6.6s %-8.8s %-8.8s %s"
+                % ("JOBID", "STATE", "TASKS", "NAME", "USER", "TIME")
```

Comment:
FWIW, this should fix the alignment issues that @grondo pointed out
```python
fmt.format(
id="JOBID",
state="STATE",
task_count="TASKS",
job_name="NAME",
username="USER",
runtime_fsd="TIME"
)
```

Reply:
ahh good idea, i'm clearly not thinking pythonic :-)